snap/squashfs/squashfs.go

```diff
@@ -110,6 +113,11 @@ func (s *Snap) Install(targetPath, mountDir string, opts *snap.InstallOptions) (
 		}
 	}
 
+	if opts.SkipOnDiskSnapValidation && osutil.FileExists(targetPath) {
+		didNothing := true
+		return didNothing, nil
+	}
+
 	// nothing to do, happens on e.g. first-boot when we already
```

Comment:
I can't add a comment below, so I'll add it here instead :-)

Below there's this:
```
	if opts == nil || !opts.MustNotCrossDevices {
```
but with your changes, the check on `opts` is no longer needed.